src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,236 @@
+    if (selectedCentury.includes(century) && selectedCentury.length === 5) {
+      const filtered = selectedCentury.filter((number) => {
+        return number === century;
+      });
+
+      setSelectedCentury(filtered);
+    } else if (!selectedCentury.includes(century)
+    && selectedCentury.length !== 5) {
+      setSelectedCentury([...selectedCentury, century]);
+    } else if (selectedCentury.includes(century)) {
+      const filtered = selectedCentury.filter((number) => {
+        return number !== century;
+      });
+
+      setSelectedCentury(filtered);
+    }
+  };
+
+  const updatedPeople = people.map((child) => {
+    if (!child.motherName && !child.fatherName) {
+      return {
+        ...child,
+        motherName: '-',
+        fatherName: '-',
+      };
+    }
+
+    if (!child.fatherName) {
+      return {
+        ...child,
+        fatherName: '-',
+      };
+    }
+
+    if (!child.motherName) {
+      return {
+        ...child,
+        motherName: '-',
+      };
+    }
+
+    const father = people.find((parent) => parent.name === child.fatherName);
+    const mother = people.find((parent) => parent.name === child.motherName);
+
+    return {
+      ...child,
+      father,
+      mother,
+    };
+  });
+
+  const filteredPeople: NewPerson[] = useMemo(
+    () => updatedPeople.filter((person) => {
+      const functionality
+       = person.name.toLowerCase().includes(query.toLowerCase().trim());
+      const centurie = Math.floor(person.born / 100);
+
+      switch (sexFilter) {
+        case FilterType.All:
+        default:
+          return functionality && selectedCentury.includes(centurie);
+        case FilterType.Male:
+          return (
+            person.sex === 'm'
+            && functionality
+            && selectedCentury.includes(centurie)
+          );
+        case FilterType.Female:
+          return (
+            person.sex === 'f'
+            && functionality
+            && selectedCentury.includes(centurie)
+          );
+      }
+    }),
+    [sexFilter, query, loading, selectedCentury],
+  );
+
+  useEffect(() => {
+    setIsfiltered(true);
+  }, [query, sexFilter, selectedCentury]);
+
+  const filterAscOrDesc = () => {
+    return filteredPeople.sort((elem1, elem2) => {
+      switch (propName) {
+        case PropName.Name:
+        case PropName.Sex:
+          return sortOrder === SortType.asc
+            ? elem1[propName].localeCompare(elem2[propName])
+            : elem2[propName].localeCompare(elem1[propName]);
+
+        case PropName.Born:
+        case PropName.Died:
+          return sortOrder === SortType.asc
+            ? elem1[propName] - elem2[propName]
+            : elem2[propName] - elem1[propName];
+
+        default:
+          return elem1.index - elem2.index;
+      }
+    });
+  };
+
+  const filterOg = () => {
+    return filteredPeople.sort((elem1, elem2) => elem1.index - elem2.index);
+  };
+
+  useEffect(() => {
```

Comment:
Instead of having two separate functions 'filterOg' and 'filterAscOrDesc', you can have one function 'filterPeople' that handles both cases, making the code more maintainable.